### PR TITLE
Validate inputs in dashboard

### PR DESCRIPTION
**Pull request by rihorn2**
Start validation work. Will show error message in place of dashboard.

### CODE REVIEW

**Comment by marksoper** [comment]
@rihorn2 Do I recall correctly that the error messages will show up as simple text in place of the dashboard? Not much styling, etc. right?

File: visualization/dashboard/src/MLIDashboard/ExplanationDashboard.tsx [hunk 1]
```diff
@@ -200,6 +214,128 @@ export class ExplanationDashboard extends React.Component<IExplanationDashboardP
...
+                rowLength = length;
+            }
+            if (length !== rowLength) {
+                return `Inconsistent dimensions. Eval dataset has length [${length}], expected [${rowLength}]`;
+            }
+            if (length === 0) {
+                return "Eval dataset not a non-empty array";
+            }
+            const fLength = props.testData[0].length;
+            if (fLength !== featureLength) {
```
Comment (imatiach-msft) [inline visualization/dashboard/src/MLIDashboard/ExplanationDashboard.tsx]: I think some of this code that checks values could be refactored out into some small utilities that do the check and take in the name of the thing being checked, as I see very similar if statements checking the same stuff several times

File: visualization/test/test-page/App.jsx [hunk 2]
```diff
@@ -32,6 +33,7 @@ import { initializeIcons } from "@uifabric/icons";
         {label: 'irisGlobal', data: irisGlobal},
         {label: 'irisDataGlobal', data: irisDataGlobal},
         {label: 'ibmData', data: ibmData},
+        {label: 'ibmDataInconsistent', data: ibmDataInconsistent},
```
Comment (imatiach-msft) [inline visualization/test/test-page/App.jsx]: can this be added to the automated tests as well somehow, which just validate than an error is not thrown or the appropriate error is thrown?

File: visualization/dashboard/src/MLIDashboard/ExplanationDashboard.tsx (hunk 1 above)
Comment (rihorn2) [reply visualization/dashboard/src/MLIDashboard/ExplanationDashboard.tsx]: definitely, but the old dashboard is in maintenance-mode, so this is more of a bug fix than adding a new feature.

**Pull request merged**